docs/dev/libs/cep.md

```diff
@@ -246,63 +334,399 @@ pattern.where(event => ... /* some condition */).or(event => ... /* or condition
 </div>
 </div>
 
-Next, we can append further states to detect complex patterns.
-We can control the contiguity of two succeeding events to be accepted by the pattern.
+##### Conditions on Contiguity
 
-Strict contiguity means that two matching events have to be directly the one after the other.
-This means that no other events can occur in between. 
-A strict contiguity pattern state can be created via the `next` method.
+FlinkCEP supports the following forms of contiguity between events:
+
+ 1. Strict Contiguity: which expects all matching events to appear strictly the one after the other,
+ without any non-matching events in-between.
+
+ 2. Relaxed Contiguity: which simply ignores non-matching events appearing in-between the matching ones.
+ 
+ 3. Non-Deterministic Relaxed Contiguity: which further relaxes contiguity by also creating alternative
+ matches which ignore also matching events.
```

Comment:
Non-Deterministic Relaxed Contiguity is hard to understand. Maybe this is clearer:

... which further relaxes contiguity, allowing additional matches that ignore some matching events.